### PR TITLE
Wrap around gettimeofday()

### PR DESCRIPTION
**Pull request by dnav**
For easier embedded OSes ports, a lwm2m_gettime() function is provided. On POSIX compliant systems, this a just a wrapper to gettimeofday with a simpler interface.
Also modifies the lwm2m_step() API.

### CODE REVIEW

**Comment by jvermillard** [comment]
+1 timeval was a PITA on MBED

File: core/liblwm2m.h [hunk 1]
```diff
@@ -520,7 +526,7 @@ int lwm2m_configure(lwm2m_context_t * contextP, char * endpointName, char * msis
 int lwm2m_start(lwm2m_context_t * contextP);
 
 // check if the server registrations are outdated and needs to be renewed
-int lwm2m_update_registrations(lwm2m_context_t * contextP, uint32_t currentTime, struct timeval * timeoutP);
+int lwm2m_update_registrations(lwm2m_context_t * contextP, time_t currentTime, time_t * timeoutP);
```
Comment (jvermillard) [inline core/liblwm2m.h]: why a time_t pointer here and not the value?

**Comment by bojh** [comment]
+1 for Arduino & Spark ports

File: core/liblwm2m.h [hunk 2]
```diff
@@ -505,8 +511,8 @@ lwm2m_context_t * lwm2m_init(lwm2m_connect_server_callback_t connectCallback, lw
 // close a liblwm2m context.
 void lwm2m_close(lwm2m_context_t * contextP);
 
-// perform any required pending operation and adjust timeoutP to the maximal time interval to wait.
-int lwm2m_step(lwm2m_context_t * contextP, struct timeval * timeoutP);
+// perform any required pending operation and adjust timeoutP to the maximal time interval to wait in seconds.
+int lwm2m_step(lwm2m_context_t * contextP, time_t * timeoutP);
```
Comment (jvermillard) [inline core/liblwm2m.h]: Same: why a time_t pointer here and not the value?

Comment (dnav) [inline core/liblwm2m.h]: Because lwm2m_step modify the timeout. (timeout is an OUT parameter).

File: core/liblwm2m.h (hunk 1 above)
Comment (dnav) [inline core/liblwm2m.h]: Actually, this function should update the timeout if a registration is needed. I will change this

**Comment by eclipsewebmaster** [comment]
:white_check_mark: 
#### Eclipse validation results:
All committers passed Eclipse CLA and Signed-off-by validation.

**Comment by dnav** [comment]
Updated.
I also removed lwm2m_update_registration**s**() from the exposed APIs. I could not figure out why it was here.

**Pull request merged**